virtio-devices/src/transport/pci_device.rs

```diff
@@ -882,11 +882,20 @@ impl PciDevice for VirtioPciDevice {
         let device = device_clone.lock().unwrap();
 
         let mut settings_bar_addr = None;
+        let mut use_64bit_bar = self.use_64bit_bar;
         if let Some(resources) = &resources {
             for resource in resources {
-                if let Resource::PciBar { index, base, .. } = resource {
+                if let Resource::PciBar {
+                    index, base, type_, ..
+                } = resource
+                {
                     if *index == VIRTIO_COMMON_BAR_INDEX {
                         settings_bar_addr = Some(GuestAddress(*base));
+                        use_64bit_bar = match *type_ {
+                            PciBarType::Io => return Err(PciDeviceError::InvalidResource),
```

Comment:
You could add the resource type to the error.

Reply:
Good point!